### PR TITLE
Reading Settings: Add Featured image email setting

### LINKED ISSUE #70416: Reading Settings: Add “Feature image” newsletter setting
Image

### PR DESCRIPTION
**Pull request by ivan-ottinger**
#### Proposed Changes

* Add Featured image email setting to the new Reading settings page

Markup on 2022-12-13 at 14:20:24

TODOs:

- [x] Address TS error. Related discussion: p1670575919753569-slack-C02TCEHP3HA; Issue: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63582; Future task: https://github.com/Automattic/wp-calypso/issues/71054
- [x] Review issue with saving → We will look at this issue separately: https://github.com/Automattic/wp-calypso/issues/71071
- [x] Update the toggle copy

#### Testing Instructions

1. Run local version of Calypso
2. Navigate to `http://calypso.localhost:3000/settings/reading/<YOUR-BLOG>`
3. Ensure the Featured image setting:
  * displays the correct value (enabled / disabled - _default_)
  * can be switched and saved
4. Validate that when the toggle is enabled, the Featured image displays in the _New Post_ email template
5. Test with Simple, Atomic and Jetpack-connected website.

ℹ️ In order for the Featured image to display in the _New Post_ email template, the post needs to have the image attached. To trigger the email, the post needs to be published on a site that has email followers, or [these steps](PdDOJh-jM-p2) can be followed.

ℹ️ Be aware about this issue: https://github.com/Automattic/jetpack/issues/27763.

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70416

### CODE REVIEW

**Comment by ivan-ottinger** [comment]
I have now addressed the TS error discussed in p1670575919753569-slack-C02TCEHP3HA with a workaround commit https://github.com/Automattic/wp-calypso/commit/e57da8952ea965218560f3c7da4d6a1b60cf4af7.

Once https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63582 is fixed, the change introduced in the commit can be reverted. For that I created a new task in our project so we won't forget: https://github.com/Automattic/wp-calypso/issues/71054.

**Comment by ivan-ottinger** [comment]
Even though there might be a [small copy change coming](https://github.com/Automattic/wp-calypso/pull/70870#discussion_r1043388908), this PR is ready for review.

---

We have decided to address the "Changes you made may not be saved." dialog bug in a separate issue (as it doesn't affect Featured image toggle setting only): https://github.com/Automattic/wp-calypso/issues/71071.

As mentioned in the [previous comment](https://github.com/Automattic/wp-calypso/pull/70981#issuecomment-1346287350), the TS bug we discovered will be addressed separately as well - as it is part of a different repository.

File: client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const ToggleControl = OriginalToggleControl as React.ComponentType<
+	OriginalToggleControl.Props & {
+		disabled?: boolean;
+	}
+>;
+
+type FeaturedImageEmailSettingProps = {
+	value?: boolean;
+	handleToggle: ( field: string ) => ( ( isChecked: boolean ) => void ) | undefined;
```
Comment (mashikag) [inline client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx:16]: No biggie but it is more common to use `?` than to use `| undefined` to make a return value type optional.

**Review by mashikag — APPROVED** [review]
The code looks neat and the changes seem to work as expected. 👍 
Screen Shot 2022-12-12 at 18 54 12

File: client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx (hunk 1 above)
Comment (mashikag) [reply client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx:16]: My bad! This is the return value of the function as in: `( field: string ) => ( ( ( isChecked: boolean ) => void ) | undefined ) `. In that case it is alright. 😅

Comment (ivan-ottinger) [reply client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx:16]: > No biggie but it is more common to use ? than to use | undefined to make a return value type optional.

Thank you for sharing that! What is not clear to me though is how to properly type a function that can return `( isChecked: boolean ) => void )` or `undefined` without using the `|` pipe operator. 🤔 

If the whole `handleToggle` could be `undefined`, then I would just put `?` there. But in the case above, what can be `undefined` is the output of that `handleToggle` function.

Comment (ivan-ottinger) [reply client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx:16]: > My bad! This is the return value of the function as in: ( field: string ) => ( ( ( isChecked: boolean ) => void ) | undefined ) . In that case it is alright. 😅

Ah! When I reloaded the page after sending my previous reply, your response appeared! :D Ok, so all clear now!

Comment (ivan-ottinger) [reply client/my-sites/site-settings/reading-newsletter-settings/FeaturedImageEmailSetting.tsx:16]: Marking this conversation as resolved. I am going for a lunch and will merge the PR once I am back. 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7670897

Thank you @ivan-ottinger for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7670897).

**Pull request merged**